src/Symfony/Component/Lock/Store/RetryTillSaveStore.php

```diff
@@ -81,7 +82,11 @@ public function waitAndSave(Key $key)
      */
     public function putOffExpiration(Key $key, $ttl)
     {
-        $this->decorated->putOffExpiration($key, $ttl);
+        try {
+            $this->decorated->putOffExpiration($key, $ttl);
+        } catch (NotExpirableStoreException $ex) {
```

Comment:
IMHO we shouldn't catch the exception here.

Reply:
I was thinking of the case where you mentioned if in DEV the application is running an `expirable` store and in production if it's changed to `non-expirable` store then this should deal with that.

Reply:
My initial point was that "a client with a Lock using a generic StoreInterface is not able to know whether the store will throw an exception or not". But now things changed: all stores  (that does not support the feature) are throwing an exception and that exception is catch by the `Lock` class. This decorator could throw an exception here

Reply:
Ahh yes that makes sense. Fixed.